wheat_filtration/mallet.py

```diff
@@ -73,42 +85,44 @@ def _make_doctopic_matrix(mallet_doctopic_filepath):
         corpus_doc_topics = in_file.readlines()
         n_docs = len(corpus_doc_topics)
         # the number of topics of the first line of the file
-        n_topics = len(corpus_doc_topics[0].split()[:2])
+        n_topics = len(corpus_doc_topics[0].split()[2:])
         doc_topic_matrix = np.zeros((n_docs, n_topics))
-        for i, line in enumerate(in_file):
+        for i, line in enumerate(corpus_doc_topics):
             topics = line.split()[2:]
-            # is it ok to set a line of a matrix with a regular array of floats?
             doc_topic_matrix[i] = [
                 float(topic_prop) for topic_prop in topics]
     return (doc_topic_matrix, n_docs, n_topics)
 
 
-def _make_mallet_model(corpus_filepath, path_to_mallet, remove_stopwords, num_topics, **kwargs):
+def _make_mallet_model(corpus_filepath, path_to_mallet, remove_stopwords, corpus_language, num_topics, **kwargs):
     """Returns a tuple containing a gensim-created topic model (class LdaMallet),
-    the preprocessed corpus documents (iter of iter of str), and the corpus vocabulary
-    (iter of str).
-    """
+    the preprocessed corpus documents (OrderedDict), and the corpus vocabulary
+    (iter of str). The keys for the document dictionary are unique document ids of
+    the format "doc<i>" where i is the number of the document in the corpus."""
     prepped_corpus = munge.corpus_to_doc_tokens(
         munge.import_corpus(corpus_filepath))
-    stop_words = stopwords.words("english")
     if remove_stopwords:
+        stop_words = stopwords.words(corpus_language)
         prepped_corpus = [
-            [word for word in doc if word not in stop_words] for doc in prepped_corpus]
+            [word.lower() for word in doc if word.lower() not in stop_words] for doc in prepped_corpus]
```

Comment:
I think we talked about this before; we should a) leave a comment here that says this is lowercasing everything, b) mention this in the docstring of this function, and c) leave a TODO comment that says this should be an option you can change (this will require changing some of the modeling logic in a few places, so we don't have to do it yet).

If you want, this might also be a fun way to practice writing an "Issue" on github.